packages/dds/legacy-dds/src/array/sharedArray.ts

```diff
@@ -444,9 +444,35 @@ export class SharedArrayClass<T extends SerializableTypeForSharedArray>
 				this.emitValueChangedEvent(moveOp, true /* isLocal */);
 				break;
 			}
-			case OperationType.toggle:
+			case OperationType.toggle: {
+				const entryId = arrayOp.entryId;
+				const liveEntry = this.getLiveEntry(entryId);
+				const isDeleted = !liveEntry.isDeleted;
+
+				// Toggling the isDeleted flag to undo the last operation for the skip list payload/value
+				liveEntry.isDeleted = isDeleted;
```

Comment:
Recommend doing the invert as you store it for easier readability
```suggestion
				const isDeleted = liveEntry.isDeleted;

				// Toggling the isDeleted flag to undo the last operation for the skip list payload/value
				liveEntry.isDeleted = !isDeleted;
```